gtfs/spec/en/reference.md

```diff
@@ -295,6 +305,70 @@ For examples that demonstrate how to specify a fare structure with [fare_rules.t
+
+Fares in `fare_leg_rules.txt` are queried by filtering fields until a fare cost matches the characteristics of the leg. Undefined values in the fields that are filtered will be matched by default to empty values for that field. For example, a rider taking a route with `network_id=network1` that is not defined as a network in `fare_leg_rules.network_id` will be matched to entries with empty `fare_leg_rules.network_id`. 
+
+It is recommended that consumers filter `fare_leg_rules.txt` by the fields that define the characteristics of travel as outputted from trip planning software. This would give the immediate capability to display the cost of a leg from within a trip planner. The fields in `fare_leg_rules.txt` that define characteristics of travel are:
+- `fare_leg_rules.network_id`
+- `fare_leg_rules.from_area_id`
+- `fare_leg_rules.to_area_id`
+- `fare_leg_rules.is_symmetrical`
+
+|  Field Name | Type | Presence | Description |
+|  ------ | ------ | ------ | ------ |
+| `leg_group_id` | ID | Optional | Identifies a group of entries in `fare_leg_rules.txt`.<br><br> Used to describe fare transfer rules between `fare_transfer_rules.from_leg_group_id` and `fare_transfer_rules.to_leg_group_id`.<br><br>Multiple entries in `fare_leg_rules.txt` may belong to the same `fare_leg_rules.leg_group_id`.<br><br>The same entry in `fare_leg_rules.txt` (not including `fare_leg_rules.leg_group_id`) must not belong to multiple `fare_leg_rules.leg_group_id`.|
+| `network_id` | ID referencing `routes.network_id` | Optional | Identifies a route network that applies for the fare leg rule.<br><br>If there are no matching `fare_leg_rules.network_id` values to the `network_id` being filtered, empty `fare_leg_rules.network_id` will be matched by default. |
+| `from_area_id` | ID referencing `areas.area_id` | Optional | Identifies a departure area.<br><br>If there are no matching `fare_leg_rules.from_area_id` values to the `area_id` being filtered, empty `fare_leg_rules.from_area_id` will be matched by default. |
+| `to_area_id` | ID referencing `areas.area_id` | Optional | Identifies an arrival area.<br><br>If there are no matching `fare_leg_rules.to_area_id` values to the `area_id` being filtered, empty `fare_leg_rules.from_area_id` will be matched by default. |
+| `is_symmetrical` | Enum | **Conditionally Required** | Indicates whether the fare leg rule may be applied for `fare_leg_rules.from_area_id` to `fare_leg_rules.to_area_id` as well as for `fare_leg_rules.to_area_id` to `fare_leg_rules.from_area_id`.<br><br>Valid options are:<br>`0` - Not symmetrical.<br>`1` - Symmetrical.<br><br>Conditionally Required:<br><br>- **Required** if either `fare_leg_rules.from_area_id` or `fare_leg_rules.to_area_id` are defined.<br>- **Forbidden** otherwise. |
+| `amount` | Non-negative currency amount | Optional | The cost of the fare for the leg. |
```

Comment:
can you give me a valid example where there's a line in fare_leg_rules.txt without amount?

Reply:
Good catch. This is an error introduced by an adaptation of the full proposal where `min_amount` and `max_amount` are considered. I will patch it here to **Required**. Note that an amount of 0.00 can be used to express a free pass.